src/openstudio_lib/HVACSystemsController.cpp

```diff
@@ -537,59 +541,72 @@ void HVACLayoutController::addLibraryObjectToModelNode(const OSItemId& itemId, m
+        if (boost::optional<model::ThermalZone> zone = object->optionalCast<model::ThermalZone>()) {
+          added = airLoop->multiAddBranchForZone(zone.get());
         }
       }
-    } else if (boost::optional<model::WaterUseEquipment> waterUseEquipment = object->optionalCast<model::WaterUseEquipment>()) {
-      if (boost::optional<model::WaterUseConnections> waterUseConnections = comp.optionalCast<model::WaterUseConnections>()) {
-        added = waterUseConnections->addWaterUseEquipment(waterUseEquipment.get());
-      }
-    } else if (boost::optional<model::WaterUseEquipmentDefinition> waterUseEquipmentDefinition =
-                 object->optionalCast<model::WaterUseEquipmentDefinition>()) {
-      if (boost::optional<model::WaterUseConnections> waterUseConnections = comp.optionalCast<model::WaterUseConnections>()) {
-        model::WaterUseEquipment waterUseEquipment(waterUseEquipmentDefinition.get());
-        waterUseEquipment.resetFlowRateFractionSchedule();
+    }
+  } else if (boost::optional<model::WaterUseEquipment> waterUseEquipment = object->optionalCast<model::WaterUseEquipment>()) {
+    if (boost::optional<model::WaterUseConnections> waterUseConnections = comp.optionalCast<model::WaterUseConnections>()) {
+      added = waterUseConnections->addWaterUseEquipment(waterUseEquipment.get());
+    }
+  } else if (boost::optional<model::WaterUseEquipmentDefinition> waterUseEquipmentDefinition =
+               object->optionalCast<model::WaterUseEquipmentDefinition>()) {
+    if (boost::optional<model::WaterUseConnections> waterUseConnections = comp.optionalCast<model::WaterUseConnections>()) {
+      model::WaterUseEquipment waterUseEquipment(waterUseEquipmentDefinition.get());
+      waterUseEquipment.resetFlowRateFractionSchedule();
 
-        added = waterUseConnections->addWaterUseEquipment(waterUseEquipment);
-      }
+      added = waterUseConnections->addWaterUseEquipment(waterUseEquipment);
     }
+  }
 
-    if (!added) {
-      if (remove) {
-        object->remove();
-      }
+  if (!added) {
+    if (remove) {
+      object->remove();
+    }
 
-      QMessageBox message(m_hvacSystemsController->hvacSystemsView());
+    QMessageBox message(m_hvacSystemsController->hvacSystemsView());
+    message.setText("The selected component is not allowed at this location.");
+    message.exec();
+    return;
+  }
 
-      message.setText("The selected component is not allowed at this location.");
+  if (object->iddObjectType() == openstudio::IddObjectType::OS_SwimmingPool_Indoor) {
 
-      message.exec();
+    FloorSurfaceDialog floorSurfaceDialog(object->cast<model::SwimmingPoolIndoor>());  // , m_hvacSystemsController->hvacSystemsView());
+    floorSurfaceDialog.exec();
+    if (floorSurfaceDialog.result() == QDialog::Accepted) {
+      // auto sf_ = floorSurfaceDialog.floorSurface();
+      // [[maybe_unused]] auto name = sf_->nameString();
+      // LOG_FREE(Debug, "temp", name);
+    } else {
+      object->remove();
```

Comment:
Where are you checking this is on the demand side?  This code is only executed if it is on the demand side?

Reply:
Ok I think I see how this works.  Seems fine to me

Reply:
It will bring schedules indeed and they'll be left in the model if it's removed, but that is the case for all other objects.

And yes, like I said, this is a post condition, only if the object was added in the correct location.